presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -989,7 +989,13 @@ public Node visitSpecialDateTimeFunction(SqlBaseParser.SpecialDateTimeFunctionCo
     @Override
     public Node visitExtract(SqlBaseParser.ExtractContext context)
     {
-        return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(context.identifier().getText().toUpperCase()));
+        String field = context.identifier().getText();
+        try {
+            return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(field.toUpperCase()));
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
This `catch` is overly broad. If any of the other calls throws an IllegalArgumentException, this will incorrectly produce an "invalid extract field" error.